odoo/sql_db.py

```diff
@@ -528,14 +591,24 @@ def __init__(self, cursor, lock):
         self._lock = lock
         self._lock.acquire()
         # in order to simulate commit and rollback, the cursor maintains a
-        # savepoint at its last commit
-        self._savepoint = "test_cursor_%s" % next(self._savepoint_seq)
-        self._cursor.execute('SAVEPOINT "%s"' % self._savepoint)
+        # savepoint at its last commit, the savepoint is created lazily
+        self._savepoint = None
+
+    def execute(self, *args, **kwargs):
+        if self._closed:
+            raise psycopg2.OperationalError('Unable to use a closed cursor.')
+
+        if not self._savepoint:
+            self._savepoint = self._cursor.savepoint(flush=False)
+
+        return self._cursor.execute(*args, **kwargs)
```

Comment:
The decorator `check` already provides the check:
```suggestion
    @check
    def execute(self, *args, **kwargs):
        if not self._savepoint:
            self._savepoint = self._cursor.savepoint(flush=False)
        return self._cursor.execute(*args, **kwargs)
```